strawberryfields/engine.py

```diff
@@ -531,161 +544,507 @@ def run(self, program, *, args=None, compile_options=None, run_options=None):
+        Returns:
+            bool: ``True`` if the connection is successful, and ``False`` otherwise
+        """
+        path = "/healthz"
+        response = requests.get(self._url(path), headers=self._headers)
+        return response.status_code == 200
+
+    def _url(self, path: str) -> str:
+        return self._base_url + path
+
+    @staticmethod
+    def _format_error_message(response: requests.Response) -> str:
+        body = response.json()
+        return "{} ({}): {}".format(
+            body.get("status_code", ""), body.get("code", ""), body.get("detail", "")
+        )
 
-        # Job either failed or is complete - in either case, clear the job queue so that the engine is
-        # ready for future jobs.
-        self.reset()
+    def __repr__(self):
+        return "<{}: token={}, host={}>".format(
+            self.__class__.__name__, self.token, self.host
+        )
 
-        if job.is_failed:
-            message = str(job.manager.http_response_data["meta"])
-            raise JobExecutionError(message)
-        elif job.is_complete:
-            job.result.manager.get()
-            return job.result.result.value
+    def __str__(self):
+        return self.__repr__()
 
-    def run(self, program, shots=1, **kwargs):
-        """Compile the given program and execute it by queuing a job in the Starship.
 
-        For the :class:`Program` instance given as input, the following happens:
+class StarshipEngine:
+    """A quantum program executor engine that that provides a simple interface for
+    running remote jobs in a synchronous or asynchronous manner.
 
-        * The Program instance is compiled for the target backend.
-        * The compiled program is sent as a job to the Starship
-        * The measurement results of each subsystem (if any) are stored in the :attr:`~.BaseEngine.samples`.
-        * The compiled program is appended to self.run_progs.
-        * The queued or completed jobs are appended to self.jobs.
+    **Example:**
 
-        Finally, the result of the computation is returned.
+    The following example instantiates an engine with the default configuration, and
+    runs jobs both synchronously and asynchronously.
+
+    .. code-block:: python
+
+        engine = StarshipEngine("chip2")
+
+        # Run a job synchronously
+        result = engine.run(program, shots=1)
+        # (Engine blocks until job is completed)
+        result  # [[0 1 0 2 1 0 0 0]]
+
+        # Run a job synchronously, but cancel it before it is completed
+        result = engine.run(program, shots=1)
+        ^C # KeyboardInterrupt cancels the job
+
+        # Run a job asynchronously
+        job = engine.run_async(program, shots=1)
+        job.status  # <JobStatus.QUEUED: 'queued'>
+        job.result  # InvalidJobOperationError
+        # (After some time...)
+        job.refresh()
+        job.status  # <JobStatus.COMPLETED: 'complete'>
+        job.result  # [[0 1 0 2 1 0 0 0]]
```

Comment:
This might be easier to read using Python prompt syntax. Eg:

```rest
Run a job synchronously:

>>> engine = StarshipEngine("chip2")
>>> result = engine.run(program, shots=1) # blocking call
>>> result
[[0 1 0 2 1 0 0 0]]

Run a job synchronously, but cancel it before it is completed
using a keyboard interrupt (`ctrl+c`):

>>> result = engine.run(program, shots=1)
^C---------------------------------------------------------------------------
KeyboardInterrupt                         Traceback (most recent call last)
<ipython-input-4-f1a1495c6d9c> in <module>()
----> 1 time.sleep(10)

Run a job asynchronously:

>>> job = engine.run_async(program, shots=1)
>>> job.status 
<JobStatus.QUEUED: 'queued'>
>>> job.result
InvalidJobOperationError
>>> job.refresh()
>>> job.status 
<JobStatus.COMPLETED: 'complete'>
>>> job.result
[[0 1 0 2 1 0 0 0]]
```

The nice thing about using prompts is that Sphinx renders it automatically without needing to specify a code-block or indent

Reply:
Oh nice! I'll update all the relevant docstrings to use the prompt syntax. (I assume these won't be run as doctests?)

Reply:
The docstrings for `StarshipEngine` and `Connection` have been updated to use the prompt syntax.